test/specs/modules/Sticky/Sticky-test.js

```diff
@@ -0,0 +1,205 @@
+    )
+
+    scrollAfterTrigger(wrapper, contextEl, { bottomOffset, offset, height })
+    scrollToContextBottom(wrapper, contextEl, { offset, height })
+    scrollToTop(wrapper, contextEl, { height, bottomOffset, offset })
+    scrollAfterTrigger(wrapper, contextEl, { bottomOffset, offset, height })
+    // Component should stick again to the top
+    wrapper.childAt(1).props().style.should.have.property('position', 'fixed')
+    wrapper.childAt(1).props().style.should.have.property('top', offset)
+  })
+
+  it('should return true if oversized', () => {
+    const offset = 20
+    const bottomOffset = 15
+    const height = 100000
+    const contextEl = { getBoundingClientRect: () => ({}) }
+    const wrapper = mount(
+      <Sticky context={contextEl} bottomOffset={bottomOffset} offset={offset} pushing />
+    )
+    scrollAfterTrigger(wrapper, contextEl, { bottomOffset, offset, height })
+    wrapper.instance().oversized().should.be.equal(true)
+  })
+
+  it('should fire onStick callback', () => {
+    const offset = 5
+    const bottomOffset = 10
+    const height = 50
+    const contextEl = { getBoundingClientRect: () => ({}) }
+    const fn = sandbox.spy()
+    const wrapper = mount(
+      <Sticky context={contextEl} offset={offset} bottomOffset={bottomOffset} onStick={fn} />
+    )
+
+    scrollAfterTrigger(wrapper, contextEl, { bottomOffset, offset, height })
+    fn.should.have.been.calledOnce()
+
+    scrollToTop(wrapper, contextEl, { bottomOffset, offset, height })
+    fn.should.have.been.calledOnce()
+  })
+
+  it('should fire onUnstick callback', () => {
+    const offset = 5
+    const bottomOffset = 10
+    const height = 50
+    const contextEl = { getBoundingClientRect: () => ({}) }
+    const fn = sandbox.spy()
+    const wrapper = mount(
+      <Sticky context={contextEl} offset={offset} bottomOffset={bottomOffset} onUnstick={fn} />
+    )
+
+    scrollAfterTrigger(wrapper, contextEl, { bottomOffset, offset, height })
+    fn.should.not.have.been.called()
+
+    scrollToTop(wrapper, contextEl, { bottomOffset, offset, height })
+    fn.should.have.been.calledOnce()
+  })
+
+  it('should fire onTop callback', () => {
+    const offset = 5
+    const bottomOffset = 10
+    const height = 50
+    const contextEl = { getBoundingClientRect: () => ({}) }
+    const fn = sandbox.spy()
+    const wrapper = mount(
+      <Sticky context={contextEl} offset={offset} bottomOffset={bottomOffset} onTop={fn} />
+    )
+
+    scrollAfterContext(wrapper, contextEl, { bottomOffset, offset, height })
+    fn.should.not.have.been.called()
+
+    scrollToTop(wrapper, contextEl, { bottomOffset, offset, height })
+    fn.should.have.been.calledOnce()
+  })
+
+  it('should fire onBottom callback', () => {
+    const offset = 5
+    const bottomOffset = 5
+    const height = 5
+    const contextEl = { getBoundingClientRect: () => ({}) }
+    const fn = sandbox.spy()
```

Comment:
Let's rename spies definition, e.g. spy for the `onBottom` should be named `onBottom`:

```js
const onBottom = sandbox.spy()
```